[2025-07-24T06:03:23.993+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: bis_pipeline_dag.collect_bronze scheduled__2025-07-24T04:00:00+00:00 [queued]>
[2025-07-24T06:03:23.998+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: bis_pipeline_dag.collect_bronze scheduled__2025-07-24T04:00:00+00:00 [queued]>
[2025-07-24T06:03:23.998+0000] {taskinstance.py:2170} INFO - Starting attempt 2 of 2
[2025-07-24T06:03:24.035+0000] {taskinstance.py:2191} INFO - Executing <Task(BashOperator): collect_bronze> on 2025-07-24 04:00:00+00:00
[2025-07-24T06:03:24.038+0000] {standard_task_runner.py:60} INFO - Started process 259 to run task
[2025-07-24T06:03:24.041+0000] {standard_task_runner.py:87} INFO - Running: ['airflow', 'tasks', 'run', 'bis_pipeline_dag', 'collect_bronze', 'scheduled__2025-07-24T04:00:00+00:00', '--job-id', '5', '--raw', '--subdir', 'DAGS_FOLDER/bis_pipeline_dag.py', '--cfg-path', '/tmp/tmpi_6saj27']
[2025-07-24T06:03:24.043+0000] {standard_task_runner.py:88} INFO - Job 5: Subtask collect_bronze
[2025-07-24T06:03:24.112+0000] {task_command.py:423} INFO - Running <TaskInstance: bis_pipeline_dag.collect_bronze scheduled__2025-07-24T04:00:00+00:00 [running]> on host e9ea48abe9b7
[2025-07-24T06:03:24.204+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='admin' AIRFLOW_CTX_DAG_ID='bis_pipeline_dag' AIRFLOW_CTX_TASK_ID='collect_bronze' AIRFLOW_CTX_EXECUTION_DATE='2025-07-24T04:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-07-24T04:00:00+00:00'
[2025-07-24T06:03:24.205+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-07-24T06:03:24.206+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', 'python /producer/api_producer.py']
[2025-07-24T06:03:24.211+0000] {subprocess.py:86} INFO - Output:
[2025-07-24T06:03:24.283+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-07-24T06:03:24.283+0000] {subprocess.py:93} INFO -   File "/producer/api_producer.py", line 3, in <module>
[2025-07-24T06:03:24.283+0000] {subprocess.py:93} INFO -     from kafka import KafkaProducer
[2025-07-24T06:03:24.283+0000] {subprocess.py:93} INFO - ModuleNotFoundError: No module named 'kafka'
[2025-07-24T06:03:24.293+0000] {subprocess.py:97} INFO - Command exited with return code 1
[2025-07-24T06:03:24.301+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/bash.py", line 212, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-07-24T06:03:24.303+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=bis_pipeline_dag, task_id=collect_bronze, execution_date=20250724T040000, start_date=20250724T060323, end_date=20250724T060324
[2025-07-24T06:03:24.346+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 5 for task collect_bronze (Bash command failed. The command returned a non-zero exit code 1.; 259)
[2025-07-24T06:03:24.353+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-07-24T06:03:24.369+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
